units/patch_intent.go

```diff
@@ -510,25 +513,41 @@ func setTasksToPreviousFailed(patchDoc, previousPatch *patch.Patch, project *mod
 	return nil
 }
 
-func (j *patchIntentProcessor) setToPreviousPatchDefinition(patchDoc *patch.Patch, project *model.Project, failedOnly bool) (string, error) {
-	previousPatch, err := patch.FindOne(patch.MostRecentPatchByUserAndProject(j.user.Username(), project.Identifier))
-	if err != nil {
-		return "", errors.Wrap(err, "querying for most recent patch")
-	}
-	if previousPatch == nil {
-		return "", errors.Errorf("no previous patch available")
+// setToPreviousPatchDefinition sets the tasks/variants based on a previous patch.
+// If failedOnly is set, we only use the tasks/variants that failed.
+// If patchId isn't set, we just use the most recent patch for the project.
+func (j *patchIntentProcessor) setToPreviousPatchDefinition(patchDoc *patch.Patch,
+	project *model.Project, patchId string, failedOnly bool) (string, error) {
+	var reusePatch *patch.Patch
+	var err error
+	if patchId == "" {
+		reusePatch, err = patch.FindOne(patch.MostRecentPatchByUserAndProject(j.user.Username(), project.Identifier))
+		if err != nil {
+			return "", errors.Wrap(err, "querying for most recent patch")
+		}
+		if reusePatch == nil {
+			return "", errors.Errorf("no previous patch available")
+		}
+	} else {
+		reusePatch, err = patch.FindOneId(patchId)
+		if err != nil {
+			return "", errors.Wrap(err, "querying for most recent patch")
```

Comment:
error message should be something like "querying for patch '%s'" instead of most recent i think